### ISSUE TITLE
Feature request: print subcase when an exception is thrown inside one

### ISSUE THREAD

**Issue by blastrock** [onqtam/doctest#136]
## Description

If an exception is thrown in a SUBCASE, the logged error only contain the TEST_CASE name.

This might be related to #125 

### Steps to reproduce

Here's a simple program:

```cpp
#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
#include "doctest/doctest.h"

TEST_CASE("tc")
{
    SUBCASE("ab")
    {
        std::cout << "ab" << std::endl;
    }
    SUBCASE("ab, c")
    {
        throw std::runtime_error("fail");
    }
}
```

The output is

```
ab
===============================================================================
test.cpp:4:
TEST CASE:  tc

test.cpp:4: ERROR: test case THREW exception: fail

===============================================================================
```

There is no information about which SUBCASE failed.

### Extra information

SUBCASEs seems to be implemented as ifs, I'm not sure this feature can be implemented in such a way. Maybe SUBCASEs should actually set up a lambda, but that means the user needs to add a ';' at the end of each SUBCASE. Also it wouldn't work with C++ < 11.

---

* doctest version: master **cd1d747**
* Operating System: **debian unstable**
* Compiler+version: **clang 5.0.0**

**Comment by onqtam** [comment]
This seems reasonable! And indeed it is connected to #125 
It can be implemented in C++98 without any troubles. Perhaps I'll look into fixing both issues!

**Comment by onqtam** [comment]
I haven't made progress in this regard with the default console reporter but the just released version 2.3 supports xml as the output and there you will get a much better context of where the exception was thrown from.

Anyway this is still a deficit of the console reporter so this issue will remain open until I fix that.

**Comment by onqtam** [comment]
fixed! releasing version 2.3.5 right now.

**Comment by Scorbutics** [comment]
Just a message to thank you a lot for having fixed this.
I have around 400 test cases, which are mostly subcases because they are ideal to avoid duplicating function calls, so until now I've had a lot of troubles to track where exceptions came from.
Now it will be a lot easier.
